File: sdk/go/examples/test_sdk.go
package main

import (
    "context"
    "fmt"
    "log"
    "time"
    
    "github.com/hyperliquid-labs/hyperliquid-go-sdk/client"
    "github.com/hyperliquid-labs/hyperliquid-go-sdk/types"
    "github.com/hyperliquid-labs/hyperliquid-go-sdk/websocket"
)

func main() {
    fmt.Println("🚀 Testing Hyperliquid Go SDK")
    fmt.Println("================================")
    
    // Create testnet client (no real private key needed for info endpoints)
    c := client.NewTestnetClient("")
    ctx := context.Background()
    
    // Test 1: Get all mid prices
    fmt.Println("\n📊 Test 1: Fetching Market Prices...")
    mids, err := c.Info().GetAllMids(ctx)
    if err != nil {
        log.Printf("Error getting mids: %v", err)
    } else {
        fmt.Printf("✅ Successfully fetched %d market prices\n", len(mids))
        if btcPrice, ok := mids["BTC"]; ok {
            fmt.Printf("   BTC Price: %s\n", btcPrice)
        }
    }
    
    // Test 2: Get order book
    fmt.Println("\n📖 Test 2: Fetching Order Book...")
    book, err := c.Info().GetL2Book(ctx, "BTC")
    if err != nil {
        log.Printf("Error getting order book: %v", err)
    } else {
        fmt.Printf("✅ Order book fetched for %s\n", book.Coin)
        fmt.Printf("   Levels: %d\n", len(book.Levels))
    }
    
    // Test 3: WebSocket connection
    fmt.Println("\n🔌 Test 3: Testing WebSocket Connection...")
    ws := websocket.NewManager(client.TestnetWS)
    
    err = ws.Connect(ctx)
    if err != nil {
        log.Printf("Error connecting to WebSocket: %v", err)
    } else {
        fmt.Println("✅ WebSocket connected successfully")
        
        // Subscribe to trades
        subID, err := ws.SubscribeToTrades("BTC", func(trades []types.TradeData) error {
            fmt.Printf("   Trade: %+v\n", trades[0])
            return nil
        })
        
        if err == nil {
            fmt.Printf("✅ Subscribed to BTC trades (ID: %s)\n", subID)
            
            // Listen for 5 seconds
            time.Sleep(5 * time.Second)
            
            // Unsubscribe
            ws.Unsubscribe(subID)
            fmt.Println("✅ Unsubscribed from trades")
        }
        
        ws.Disconnect()
    }
    
    fmt.Println("\n✨ Go SDK tests completed!")
}

File: sdk/java/examples/TestSDK.java
package examples;

import com.hyperliquid.sdk.client.HyperliquidClient;
import com.hyperliquid.sdk.types.*;
import reactor.core.publisher.Mono;
import java.math.BigDecimal;
import java.time.Duration;
import java.util.concurrent.CompletableFuture;

public class TestSDK {
    public static void main(String[] args) {
        System.out.println("🚀 Testing Hyperliquid Java SDK");
        System.out.println("=================================");
        
        try {
            // Create testnet client
            HyperliquidClient client = HyperliquidClient.testnet("");
            
            // Test 1: Connect to WebSocket
            System.out.println("\n🔌 Test 1: WebSocket Connection...");
            CompletableFuture<Void> connectFuture = client.connect();
            connectFuture.get();
            System.out.println("✅ WebSocket connected successfully");
            
            // Test 2: Get user state (example)
            System.out.println("\n👤 Test 2: Testing Info API...");
            String testAddress = "0x0000000000000000000000000000000000000000";
            
            Mono<UserState> userStateMono = client.info()
                .getUserState(testAddress)
                .timeout(Duration.ofSeconds(5));
            
            userStateMono.subscribe(
                state -> {
                    System.out.println("✅ User state retrieved");
                    System.out.println("   Account Value: " + state.getAccountValue());
                },
                error -> {
                    System.out.println("ℹ️ User state not available (expected for test address)");
                }
            );
            
            // Test 3: Subscribe to market data
            System.out.println("\n📊 Test 3: Market Data Subscription...");
            client.websocket().subscribeToAllMids(mids -> {
                System.out.println("✅ Received market prices update");
                System.out.println("   Markets: " + mids.size());
                return null;
            });
            
            // Keep running for demo
            Thread.sleep(5000);
            
            // Disconnect
            client.disconnect().get();
            System.out.println("\n✅ Disconnected successfully");
            
            System.out.println("\n✨ Java SDK tests completed!");
            
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
